src/filelock/_api.py

```diff
@@ -76,12 +77,41 @@ class ThreadLocalFileContext(FileLockContext, local):
 class BaseFileLock(ABC, contextlib.ContextDecorator):
     """Abstract base class for a file lock object."""
 
-    def __init__(
+    _instances: ClassVar[WeakValueDictionary[str, BaseFileLock]] = WeakValueDictionary()
+
+    def __new__(  # noqa: PYI034
+        cls,
+        lock_file: str | os.PathLike[str],
+        is_singleton: bool = False,  # noqa: FBT001, FBT002
```

Comment:
do we need default here?

Reply:
Yes, if I drop the default then the argument must be specified every time we do `FileLock(...)` unfortunately.